### DISCUSSION TITLE
SSG without `_payload.json` files

### DISCUSSION THREAD

**Discussion by septatrix** [nuxt/nuxt#26873] [Questions] [answered]
Hey everyone, I am trying to use Nuxt as a SSG and the majority (77%) of the final output size is due to `_payload.json` file in the final build. This is a problem as I am reaching the storage limit of github pages.

Based on my research the files seem to be for `useFetch` but as my site is fully static and does not use that data I have no need for it. Simply deleting the files, however, leads to errors as now fetch fails.

What I considered so far:
* I also read that it might be possible to reduce the size of these files by using the `pick` option though that is way too cumbersome if I want to implement this all over the place and does not work that well for nested objects.
* Furthermore some of these files are duplicated as they seem to be stored for each and every page and not once for every endpoint called. So one way to reduce the size is to get rid of the duplicates and symlink these files though that is far from ideal and a PITA to automate.
* Switching to AstroJS. Astro seems to properly bake the generated files and does not depend on any kind of `_payload.json` files. However, that would mean rewriting several days of work - doable but I would like to avoid doing so.

Things which might be possible but I do not know enough about them:
* Maybe some of the options to `useFetch` would prevent it from trying to fetch new data on static sites?
* Play around with `<ClientIsland>`. This seems to be quite new to Nuxt, however, I do not quite understand how to apply it, yet, and currently do not have the disposable time to play around with it.

**Comment by danielroe** [comment]
You can disable payload extraction like this:

```ts
export default defineNuxtConfig({
  experimental: {
    payloadExtraction: false
  }
})
```

But yes, you definitely should not store more data in a payload than needed - please use `pick` or `transform` to ensure that you are only storing the data required to render a page.

**Reply by septatrix** [reply]
I assume the experimental option `sharedPrerenderData` should also be usable in my situation? (It seems like there is little documentation of the prerender/payload functionality.)

> But yes, you definitely should not store more data in a payload than needed - please use `pick` or `transform` to ensure that you are only storing the data required to render a page.

While this is certainly possible it is currently not worth the effort as the data structure is quite complex. Furthermore this would mean that changes in a child component allowing it to consume more keys would result in having to change the fetch calls everywhere this component is used.

**Reply by septatrix** [reply]
> You can disable payload extraction like this

This seems to just embed it inside the page itself so does not reduce the size. What I actually want seem to be server components/pages.

**Reply by septatrix** [reply, marked as answer]
Having the payloads extracted and then removing them seems to be fine. I also had to remove `_nuxt/builds/meta/<some uuid>.json` and then it seemed to work fine...

**Reply by florian-strasser** [reply]
Seems like a still relevant problem, I have this with most of the pages I build because they simply have a specific amount of content. Doesn't matter if I only request necessary fields from database.

I wonder for what the payload is used in first place since I can't find any problems when payload is disabled in SSR, just results in better loading time...

**Reply by florian-strasser** [reply]
I would recommend to disable payload if `ssr:true` is set

**Comment by septatrix** [comment]
Not strictly related but for anyone who also tries to minimize the output size: #21821 currently results in significantly larger outputs if one has a lot of small pages
